server/service/orbit.go

```diff
@@ -249,15 +249,13 @@ func (svc *Service) GetOrbitConfig(ctx context.Context) (fleet.OrbitConfig, erro
 				notifs.RunSetupExperience = true
 			}
 
-			if inSetupAssistant || fleet.IsNotFound(err) {
-				// If the client is running a fleetd that doesn't support setup experience, or if no
-				// software/script has been configured for setup experience, then we should fall back to
-				// the "old way" of releasing the device. We do an additional check for
-				// !inSetupAssistant to prevent enqueuing a new job every time the /config
-				// endpoint is hit.
+			if inSetupAssistant {
```

Comment:
This reverts the endpoint to as it was before, since now it only enqueues the worker-based device release if the host a) has setup experience items to process and b) runs an old fleetd version that doesn't support processing setup experience stuff.

The case where there are no setup experience items is fully handled by the TokenUpdate code and uses the worker. `GetHostAwaitingConfiguration` will only return a row (with `true`) if there are setup experience items.

Reply:
This also means we can leave the deletion of the host awaiting config table with the deletion of the host, as we don't give a special meaning to the "no row present" case (we only care about it in this endpoint, and if the value is true).